docs/decisions/0013-custom-groups-for-enterprise-learners.rst

```diff
@@ -0,0 +1,152 @@
+
+Outputs
+==========
+The root URL for getting the basic information about the group
+::
+   {
+        'group_uuid': 'group_uuid',
+        'name': 'group_name',
+        'enterprise_customer_uuid': 'enterprise_customer_uuid',
+        'policy_uuid': 'policy_uuid'
+   }
+
+
+**api/v1/enterprise-group/?learner_uuid=[enterprise_customer_user_id]&enterprise_uuid=[enterprise_uuid]**
+
+Accepted Query Params
+==========
+- ``learner_uuid`` (optional): Get all the groups that the learner is associated with 
+- ``enterprise_uuid`` (optional): Get all the groups under the enterprise
+
+Outputs
+==========
+Returns a paginated list of groups filtered by the query params
+::
+   {
+       'count': 1,
+       'next': null,
+       'previous': null,
+       'results': [
+           {
+               'group_uuid': 'group_uuid',
+               'name': 'group_name',
+               'enterprise_customer_uuid': 'enterprise_customer_uuid',
+               'policy_uuid': 'policy_uuid'
+           }
+       ]
+   }
+
+
+**GET (list) /learners**
+------
+**api/v1/enterprise-group/<group_uuid>/learners/**
+
+Outputs
+==========
+Returns a paginated list of learners that are associated with the enterprise group uuid 
+::
+   {
+       'count': 1,
+       'next': null,
+       'previous': null,
+       'results': [
+           {
+               'learner_uuid': 'enterprise_customer_user_id',
+               'enterprise_group_membership_uuid': 'enterprise_group_membership_uuid',
+           }
+       ]
+   }
+
+
+**POST /assign_learners**
+------
+**api/v1/enterprise-group/<group_uuid>/assign_learners**
+
+Inputs
+==========
+- ``learner_uuids`` (POST data, required): A list of enterprise_customer_user_ids to assign to the group
+
+Outputs
+==========
+Returns a list of the EnterpriseGroupMembership objects that were created 
+::
+   {
+       'count': 1,
+       'next': null,
+       'previous': null,
+       'results': [
+           {
+               'learner_uuid': 'enterprise_customer_user_id',
+               'enterprise_group_membership_uuid': 'enterprise_group_membership_uuid',
+           }
+       ]
+   }
+
+
+**POST /remind_learners**
+------
+**api/v1/enterprise-group/<group_uuid>/remind_learners**
+
+Inputs
+==========
+- ``learner_uuids`` (POST data, required): A list of enterprise_customer_user_ids to remind of their group membership
+
+
+**POST /remove_learners**
+------
+**api/v1/enterprise-group/<group_uuid>/remove_learners**
+
+Inputs
+==========
+- ``learner_uuids`` (POST data, required): A list of enterprise_customer_user_ids to assign to the group
+
+   
+EnterpriseGroupMembership
+*********************
+**Model properties**
+------
+- uuid, created, modified, history (boilerplate)
+- group (NOT NULL, FK to EnterpriseGroup with related name ``members``)
+- enterprise_customer_user_id (NOT_NULL, FK to EnterpriseCustomerUser with related_name of ``memberships``)
+- status (NOT NULL, char selection: (`PENDING`, `ACCEPTED`, `EXPIRED`, `REMOVED`))
```

Comment:
imho we shouldnt include any of this complexity in V1. learners are assigned to a group without having to accept that invitation (in the mocks ive seen thus far). this table also doesnt really need its own PK - the group_uuid, lms_user_id pair is enough to address into a specific membership - unless django ORM requires this but if its just a join table (drop the remind, status stuff) this shouldnt need to be too complicated.

Reply:
We should talk to product because it was my assumption that pending group status was a requirement for MVP